addons/point_of_sale/static/src/js/models.js

```diff
@@ -2005,7 +2005,14 @@ exports.Orderline = Backbone.Model.extend({
             var tax = _.detect(taxes, function(t){
                 return t.id === el;
             });
-            product_taxes.push.apply(product_taxes, self._map_tax_fiscal_position(tax));
+            _(self._map_tax_fiscal_position(tax)).each(function(fiscal_position_tax) {
+                var tax_index = product_taxes.findIndex(function(product_tax) {
+                    return product_tax.id === fiscal_position_tax.id;
+                })
+                if (tax_index === -1) {
+                    product_taxes.push(fiscal_position_tax);
+                }
+            });
```

Comment:
Maybe we can shorten with just 
```diff
diff --git a/addons/point_of_sale/static/src/js/models.js b/addons/point_of_sale/static/src/js/models.js
index e76064d82d4..02bc0db4745 100644
--- a/addons/point_of_sale/static/src/js/models.js
+++ b/addons/point_of_sale/static/src/js/models.js
@@ -2006,6 +2006,7 @@ exports.Orderline = Backbone.Model.extend({
                 return t.id === el;
             });
             product_taxes.push.apply(product_taxes, self._map_tax_fiscal_position(tax));
+            product_taxes = [...new Set(product_taxes)];
         });
 


```
@nle-odoo What do you think?

Reply:
these are objects I think so I don't think they will be equals in Set

but we if we want to remove any duplicate, we could probably do (one time after the loop):

```py
product_taxes = _.uniq(product_taxes, function (tax) { return tax.id; });
```

Reply:
Yes. The last proposition is working just fine. :ok_hand: